doc/concepts/pipeline-concepts/pipeline/index.rst

```diff
@@ -0,0 +1,61 @@
+Pipeline
+========
+
+A pipeline is a Pachyderm primitive that is responsible for collecting data
+from a specified source, or ``pfs/input``, transform it according to the pipeline
+configuration file, and place it to the ``pfs/out`` directory in the ``pachd``
+container. Each pipeline subscribes to a branch in the
+input repository or repositories, gets triggered by a commit,
+and runs your code to completion.
+
+A minimum pipeline specification must include the following parameters:
+
+- ``name`` — The name of your data pipeline. You can set an arbitrary
```

Comment:
> The name of your data pipeline. You can set an arbitrary
  name that is meaningful to the code you want to run to process the
  data in the specified repository.

The name of your data pipeline. You should give your pipeline a meaningful
name as it will write all output results to a repo of the same name. Pipeline names can be... <-- insert constraints on pipeline names. You'll need to figure that out from eng. There is a max character limit (something like 50ish characters), and I think i can be alphanumeric and include dash or underscore. But also, I think dashes might break S3gateway or something like that, so truthfully, we should make sure the actual code get updated there and then also document it correctly.